src/core/useAtom.ts

```diff
@@ -86,12 +95,14 @@ export function useAtom<Value, Update, Result extends void | Promise<void>>(
   )
 
   if (atomFromUseReducer !== atom) {
-    dispatch(undefined)
+    rerenderIfChanged(undefined)
   }
 
   useEffect(() => {
-    const unsubscribe = store[SUBSCRIBE_ATOM](atom, dispatch)
-    dispatch(undefined)
+    // Call `rerenderIfAtomStateChanged` whenever this atom is invalidated. Note
```

Comment:
```suggestion
    // Call `rerenderIfChanged` whenever this atom is invalidated. Note
```

btw, I wonder if there could be a better name than `rerenderIfChanged`. Seems good for now.

Reply:
I agree that the name could be better. I tried a few iterations (like `rerenderIfAtomStateChanged`), but I was also trying to avoid Prettier from reflowing the entire callback definition due to line length.

Reply:
Yeah, good point about prettier.